modules/supervisord.nix

```diff
@@ -94,7 +94,9 @@ let
       autorestart = program.autoRestart;
       environment = let
         # FIXME: Make more robust
-        escape = builtins.replaceStrings [ "%" ] [ "%%" ];
+        escape = s:
+          assert lib.assertMsg (!(lib.hasInfix "\"" s)) "supervisord.programs.<name>.environment: Values cannot have double quotes at the moment (${s})";
```

Comment:
Using asserts llike this will fail the whole evaluation and only show on assertion error at a time. Can you rather use NixOS style assertions? (https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/assertions.nix)